### PR TITLE
Auto reconnect when connection fails

### LINKED ISSUE #56: Ability to reconnect when theres a network issue
Sometimes even the DNS resolution fails. Cases like that, can reduce resilience. We can improve it if we add the ability to Snorlax try to reconnect or make the request again.

### PR DESCRIPTION
**Pull request by pedrommone**
Closes #56

### CODE REVIEW

File: src/RestClient.php [hunk 1]
```diff
@@ -293,11 +295,44 @@ public function request($method, $uri, $options = [])
...
+            return $this->client->request($method, $uri, $options);
+        } catch (ConnectException $e) {
+            if ($options['retries'] > 0) {
+                --$options['retries'];
+
+                return $this->requestHub($method, $uri, $options);
+            }
+
+            throw new ConnectException($e->getMessage(), $e->getRequest(), null, $e->getHandlerContext());
+        } catch (\GuzzleException $e) {
```
Comment (ocesaroliveira) [inline src/RestClient.php]: If you don't declare these catches, PHP will throw normally the exception

**Comment by ocesaroliveira** [comment]
You forget to cover this with tests

**Comment by lucasmezencio** [comment]
Why did you add the `retries` to all other tests?

**Comment by pedrommone** [comment]
@lucasmezencio probably to control the tests.

**Comment by lucasmezencio** [comment]
@pedrommone I don't like this idea. Can you remove them to see how the tests behave?

**Pull request merged**